object.go

```diff
@@ -578,77 +613,82 @@ func (obs *obs) Delete(name string) error {
 		return ErrBadObjectMeta
 	}
 
-	// Place a rollup delete marker.
+	// Place a rollup delete marker / publish the meta
 	info.Deleted = true
 	info.Size, info.Chunks, info.Digest = 0, 0, _EMPTY_
-
-	metaSubj := fmt.Sprintf(objMetaPreTmpl, obs.name, sanitizeName(name))
-	mm := NewMsg(metaSubj)
-	mm.Data, err = json.Marshal(info)
-	if err != nil {
+	if err = publishMeta(obs, name, info); err != nil {
 		return err
 	}
-	mm.Header.Set(MsgRollup, MsgRollupSubject)
-	_, err = obs.js.PublishMsg(mm)
+
+	// Purge chunks for the object
+	return purgeChunks(obs, info.NUID)
+}
+
+func doesntExistOrIsDeleted(obs *obs, name string) error {
+	info, err := obs.GetInfo(name)
 	if err != nil {
+		if err == ErrObjectNotFound {
```

Comment:
Combine into one if I think.

Reply:
I could use an example

Reply:
```go
info, err := obs.GetInfo(name)
if err == ErrObjectNotFound || (info != nil && info.Deleted) {
  return nil
} else if err != nil {
  return err
}
return errors.New("nats: name already used for an existing object")
```

Reply:
Not sure that's any better ;)

Reply:
yeah, that's about what I had. It's six of one